src/lib/web-component/filter-timestamp/filter-timestamp-element.js

```diff
@@ -0,0 +1,86 @@
+export class FilterTimestamp extends HTMLElement {
+  constructor() {
+    super();
+
+    this.timeStamp = {};
+    this.onHiddenListener = () => browser.menus.removeAll();
+
+    const shadow = this.attachShadow({ mode: 'open' });
+
+    const filterTSWrapper = document
+      .querySelector('#filterTimestampTemplate')
+      .content.cloneNode(true);
+
+    this.filterTSContainer = filterTSWrapper.querySelector(
+      '.timestamp-filter-container'
+    );
+
+    this.startTimeLabel = filterTSWrapper.querySelector('#startTimeLabel');
+    this.stopTimeLabel = filterTSWrapper.querySelector('#stopTimeLabel');
+    this.clearTSFilterBtn = filterTSWrapper.querySelector('#clearBtn');
+
+    shadow.appendChild(filterTSWrapper);
+  }
+
+  setFilterRange = (info, tab) => {
+    if (tab.id !== this.tab.id) {
+      return;
+    }
+
+    const selectedEl = browser.menus.getTargetElement(info.targetElementId);
+    const selectedRow = selectedEl.closest('tr');
+
+    if (!selectedRow) {
+      return;
+    }
+
+    const chosenTimestamp = selectedRow.querySelector('.timestamp').textContent;
+    if (info.menuItemId === 'startTime') {
+      this.timeStamp.start = Date.parse(chosenTimestamp);
+      this.startTimeLabel.textContent = chosenTimestamp;
+    } else if (info.menuItemId === 'stopTime') {
+      this.timeStamp.stop = Date.parse(chosenTimestamp);
+      this.stopTimeLabel.textContent = chosenTimestamp;
+    }
+
+    this.filterTSContainer.hidden = false;
+    this.disptachTSFilterEvent();
+  };
+
+  disptachTSFilterEvent() {
+    const filterDetail = {
+      updateFilter: {
+        timeStamp: this.timeStamp,
+      },
+    };
+
+    this.dispatchEvent(
+      new CustomEvent('filterchange', { detail: filterDetail })
+    );
+  }
+
+  onClearTSFilter = () => {
+    this.startTimeLabel.textContent = 'From Beginning';
+    this.stopTimeLabel.textContent = 'Up to End';
+    this.filterTSContainer.hidden = true;
+
+    this.timeStamp = {};
+    this.disptachTSFilterEvent();
+  };
+
+  async connectedCallback() {
+    browser.menus.onClicked.addListener(this.setFilterRange);
+    browser.menus.onHidden.addListener(this.onHiddenListener);
+    this.clearTSFilterBtn.addEventListener('click', this.onClearTSFilter);
+
+    this.tab = await browser.tabs.getCurrent();
```

Comment:
@Rob--W sigh... choosing to use the browser-.menu API for the timestamp filter context menu isn't going to work once we try to use the activitylog page inside a devtools panel...